application/src/main/java/org/opentripplanner/routing/algorithm/filterchain/filters/transit/RemoveTransitIfStreetOnlyIsBetter.java

```diff
@@ -41,10 +52,21 @@ public List<Itinerary> flagForRemoval(List<Itinerary> itineraries) {
       .mapToInt(Itinerary::getGeneralizedCost)
       .min();
 
-    if (minStreetCost.isEmpty()) {
+    if (minStreetCost.isEmpty() && streetOnlyCost.isEmpty()) {
+      // If no cost is found an empty list is returned.
       return List.of();
+    } else if (minStreetCost.isPresent() && streetOnlyCost.isPresent()) {
+      // If both the minStreetCost and streetOnlyCost are present, take the minimum value.
```

Comment:
Is this possible a case?

Reply:
This should not be a possible case since the walk direct search is removed with paging. I changed this to throw an error if this case happens